sklearn/svm/base.py

```diff
@@ -170,6 +169,8 @@ def fit(self, X, y, sample_weight=None):
                              "boolean masks (use `indices=True` in CV)."
                              % (sample_weight.shape, X.shape))
 
+        y = self._validate_targets(y, sample_weight)
```

Comment:
I must be missing something. I think we're removing elements from y but not removing the corresponding elements from X. How does that work?

Reply:
The solution is updating neither X nor y.  It is updating an existing local variable y_, used to compute self.class_weights.

The issue is, libsvm,  the underlying library, rightfully considers only the subset (X,y)[sample_weight !=0] and not full (X,y). The fact that this wrapper class, designed to augment the results of underlying library, is computing self.cls and self.class_weights using full (X,y) without mirroring the behavior of underlying library, is the bug. This fix addresses the core assumption that libsvm call (with sample weights) considers full (X,y).

Reply:
Raising an error implies that sample_weights, where a class has total weight of 0 is not supported. This means, the burden is on the caller to pre-filter the inputs, which can be annoyance in ensemble classifiers that use random sampling.  Moreover, I do not think it is a case of wrong input, so throwing an error does not make sense.